code/controllers/subsystem/jobs.dm

```diff
@@ -726,3 +726,135 @@ SUBSYSTEM_DEF(jobs)
 		new_id_change_records["[id_change_counter]"] = thisrecord
 		id_change_counter++
 	id_change_records = new_id_change_records
+
+// This proc will update all players EXP at once. It will calculate amount of time to add dynamically based on the SS fire time.
+/datum/controller/subsystem/proc/batch_update_player_exp(announce = FALSE)
+	// Right off the bat
+	var/start_time = start_watch()
+	// First calculate minutes
+	var/divider = 10 // By default, 10 deciseconds in 1 second
+	if(flags & SS_TICKER)
+		divider = 20 // If this SS ever gets made into a ticker SS, account for that
+
+	var/minutes = (wait / divider) / 60 // Calculate minutes based on the SS wait time (How often this proc fires)
+
+	// Step 1: Get us a list of clients to process
+	var/list/client/clients_to_process = GLOB.clients.Copy() // This is copied so that clients joining in the middle of this dont break things
+	log_debug("Starting EXP update for [length(clients_to_process)] clients. (Adding [minutes] minutes)")
```

Comment:
Does this get messaged to admins? If so this is gonna be 24 lines of debug logs every round. It'll get annoying.